boat_simulator/nodes/physics_engine/kinematics_computation.py

```diff
@@ -0,0 +1,100 @@
+"""This module contains the kinematics computations for the boat."""
+
+from typing import Tuple
+from numpy.typing import ArrayLike
+from boat_simulator.common import utils
+from boat_simulator.common.types import Scalar
+from boat_simulator.nodes.physics_engine.kinematics_data import KinematicsData
+import numpy as np
+import boat_simulator.common.constants as constants
+
+
+class BoatKinematics:
+    # Private class member default
+    __timestep = 1.0
+    __relative_data = KinematicsData()
+    __global_data = KinematicsData()
+
+    def __init__(self, timestep: Scalar) -> None:
+        self.__timestep = timestep
+
+    # Updates the kinematics data and returns the kinematics data objects
+    def step(self, net_force: ArrayLike, net_torque: ArrayLike
+             ) -> Tuple[KinematicsData, KinematicsData]:
+
+        next_relative_acceleration = self.__compute_next_relative_acceleration(
+            self.boat_mass, net_force
+        )
+        next_relative_velocity = self.__compute_next_relative_velocity(
+            self.relative_data.cartesian_velocity, next_relative_acceleration
+        )
+        next_relative_position = self.__compute_next_position(
+            self.relative_data.cartesian_position,
+            next_relative_velocity,
+            next_relative_acceleration
+        )
+
+        next_ang_acceleration = self.__compute_next_ang_acceleration(
+            self.inertia_inverse, net_torque
+        )
+        next_ang_velocity = self.__compute_next_ang_velocity(
+            self.relative_data.angular_velocity, next_ang_acceleration
+        )
+        next_ang_position = self.__compute_next_position(
+            self.relative_data.angular_position, next_ang_velocity, next_ang_acceleration
+        )
+
+        yaw_radians = utils.degrees_to_rad(
+            next_ang_position[constants.ORIENTATION_INDICES.YAW.value]
+        )
+
+        next_global_acceleration = next_relative_acceleration * \
+            np.array([np.sin(yaw_radians), np.cos(yaw_radians), 0])
+        next_global_velocity = next_relative_velocity * \
+            np.array([np.sin(yaw_radians), np.cos(yaw_radians), 0])
```

Comment:
I changed the [angle convention](https://ubcsailbot.atlassian.net/wiki/spaces/prjt22/pages/1809678337/Physics+Engine+Fluid+Simulation+Design+Specification#Physics-Model). All you need to do is swap sine and cosine.